miniapps/navier/cht.cpp

```diff
@@ -0,0 +1,641 @@
+      {
+         VisualizeField(vis_sol, vishost, visport, *u_gf,
+                        "Velocity", Wx, Wy, Ww, Wh);
+      }
+      else
+      {
+         VisualizeField(vis_sol, vishost, visport, *t_gf,
+                        "Temperature", Wx, Wy, Ww, Wh);
+      }
+   }
+
+   if (ode_solver) { ode_solver->Init(*coper); }
+
+   for (int step = 0; !last_step; ++step)
+   {
+      if (t + dt >= t_final - dt / 2)
+      {
+         last_step = true;
+      }
+
+      double cfl;
+      if (flowsolver)
+      {
+         flowsolver->Step(t, dt, step);
+         cfl = flowsolver->ComputeCFL(*u_gf, dt);
+      }
+      if (ode_solver)
+      {
+         ode_solver->Step(t_tdof, t, dt);
+         t_gf->SetFromTrueDofs(t_tdof);
+      }
+      finder.Interpolate(vxyz, color_array, *u_gf, interp_vals);
+      if (color == 1)
+      {
+         *u_gf = interp_vals;
+         adv_gf_c.SetGridFunction(u_gf);
+         coper->SetParameters(adv_gf_c);
+      }
+
+      if (visualization)
+      {
+         if (color == 0)
+         {
+            VisualizeField(vis_sol, vishost, visport, *u_gf,
+                           "Velocity", Wx, Wy, Ww, Wh);
+         }
+         else
+         {
+            VisualizeField(vis_sol, vishost, visport, *t_gf,
+                           "Temperature", Wx, Wy, Ww, Wh);
+         }
+      }
+
+
+      if (color == 0 && myidlocal == 0)
+      {
+         printf("%11s %11s %11s\n", "Time", "dt", "CFL");
+         printf("%.5E %.5E %.5E\n", t, dt,cfl);
+         fflush(stdout);
+      }
+   }
+
+   if (flowsolver) { flowsolver->PrintTimingData(); }
+
+   if (flowsolver)
+   {
+      ostringstream mesh_name;
+      mesh_name << "fluid.mesh";
+      ofstream mesh_ofs(mesh_name.str().c_str());
+      mesh_ofs.precision(8);
+      pmesh->PrintAsOne(mesh_ofs);
+
+      ostringstream sol_name;
+      sol_name << "fluid.gf";
+      ofstream sol_ofs(sol_name.str().c_str());
+      sol_ofs.precision(8);
+      u_gf->SaveAsOne(sol_ofs);
+   }
+   if (ode_solver)
+   {
+      ostringstream mesh_name;
+      mesh_name << "solid.mesh";
+      ofstream mesh_ofs(mesh_name.str().c_str());
+      mesh_ofs.precision(8);
+      pmesh->PrintAsOne(mesh_ofs);
+
+      ostringstream sol_name;
+      sol_name << "solid.gf";
+      ofstream sol_ofs(sol_name.str().c_str());
+      sol_ofs.precision(8);
+      t_gf->SaveAsOne(sol_ofs);
+   }
+
+   delete coper;
+   delete t_gf;
+   delete u_gf;
+   delete adv_fes_s;
+   delete fes_s;
+   delete fec_s;
+   delete ode_solver;
+   delete flowsolver;
+   delete pmesh;
+
+   return 0;
+}
+
+ConductionOperator::ConductionOperator(ParFiniteElementSpace &f, double al,
+                                       double kap,
+                                       VectorGridFunctionCoefficient adv_gf_c)
+   : TimeDependentOperator(f.GetTrueVSize(), 0.0), fespace(f), M(NULL), K(NULL),
+     T(NULL), current_dt(0.0),
+     M_solver(f.GetComm()), T_solver(f.GetComm()), z(height), udir(10)
```

Comment:
I suggest changing the ordering of the declaration of `z` and `udir` to eliminate the compiler warning below.
`cht.cpp:443:52: warning: field 'z' will be initialized after field 'udir' [-Wreorder-ctor]
     M_solver(f.GetComm()), T_solver(f.GetComm()), z(height), udir(10)`